### PR TITLE
ignore match should avoid checking absolute paths for filenames

### LINKED ISSUE #1533: Filter used by webapp happens to be filtering on Tomcat path
Hi folks,

I like the new opengrok 1.0 with the navigation pane and would like to migrate.

However I have a problem which drives me nuts.  Searching works fine. But the directory listing is always empty. Eg

http://localhost:8095/1830srosviews/xref/panos-sidewinder/iom_api/iom_api.h

fileaccess is fine. However 

http://localhost:8095/1830srosviews/xref/panos-sidewinder/iom_api/

shows only an empty file list.

I had also to upgrade the the apache-tomcat-6.5 to apache-tomcat-9.0.0.M19.

Caching is switched of.
Does anybody have the same problem? 

A1: firebug httml response protocol:

```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
    "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
<meta name="robots" content="noindex,nofollow" />
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
<meta name="generator" content="{OpenGrok 1.0 (2850968932c2933b8ebb1f9808991cda0be846ed)" />
<link rel="icon" href="/1830srosviews/default/img/icon.png" type="image/png" />
<link rel="stylesheet" type="text/css" media="all"
    title="Default" href="/1830srosviews/default/style.css" />
<link rel="alternate stylesheet" type="text/css" media="all"
    title="Paper White" href="/1830srosviews/default/print.css" />
<link rel="stylesheet" type="text/css" href="/1830srosviews/default/print.css" media="print" />
<link rel="stylesheet" type="text/css" href="/1830srosviews/default/jquery-ui-1.12.0-custom.min.css"
 />
<link rel="stylesheet" type="text/css" href="/1830srosviews/default/jquery.tooltip.css" />
<link rel="stylesheet" type="text/css" href="/1830srosviews/default/jquery.tablesorter.css" />
<link rel="stylesheet" type="text/css" href="/1830srosviews/default/searchable-option-list-2.0.2.min
.css" />
<link rel="search" href="/1830srosviews/opensearch"
    type="application/opensearchdescription+xml"
    title="OpenGrok Search for current project(s)" />
<script type="text/javascript">/* <![CDATA[ */
    window.contextPath = '/1830srosviews';
    document.xrefPath = '/1830srosviews/xref';
    document.pageReady = [];
    document.domReady = [];
/* ]]> */
</script>
<title>iom_api - OpenGrok cross reference for /panos-sidewinder/iom_api/</title></head><body>
<script type="text/javascript">/* <![CDATA[ */
    document.rev = function() { return getParameter("r"); };
    document.annotate = false;
    document.domReady.push(function() { domReadyMast(); });
    document.pageReady.push(function() { pageReadyMast(); });
/* ]]> */</script>
<div id="page">
    <div id="whole_header">
<div id="header"></div>
<div id="Masthead">
    <a href="/1830srosviews/xref/">xref</a>: /<a href="/1830srosviews/xref/panos-sidewinder/">panos-sidewinder
</a>/<a href="/1830srosviews/xref/panos-sidewinder/iom_api/">iom_api</a>/<span id="dtag">
    </span>
    </div>
<form action="/1830srosviews/search">
<div id="bar">
    <ul>
        <li><a href="/1830srosviews/"><span id="home"></span>Home</a></li><li><a href="/1830srosviews
/history/panos-sidewinder/iom_api/"><span id="history"></span>History</a></li><li><span class="c"><span
 class="annotate"></span>Annotate</span></li><li><input type="text" id="search" name="q" class="q" /
>
            <input type="submit" value="Search" class="submit" /></li><li><input type="checkbox" name
="path" value='"/panos-sidewinder/iom_api/"' /> only in <b>this directory</b></li>
    </ul><input type="hidden" id="contextpath" value="/1830srosviews" />
</div>
        </form>
    </div>
<div id="content">
<script type="text/javascript">/* <![CDATA[ */
document.pageReady.push(function() { pageReadyList();});
/* ]]> */</script>
<table id="dirlist" class="tablesorter tablesorter-default">
<thead>
<tr>
<th class="sorter-false"></th>
<th>Name</th>
<th class="sorter-false"></th>
<th class="sort-dates">Date</th>
<th class="sort-groksizes">Size</th>
</tr>
</thead>
<tbody>
<tr><td><p class="'r'"/></td><td><b><a href="..">..</a></b></td><td></td><td>Today</td><td>4 KiB</td
></tr>
</tbody>
</table><div id="footer">
<p><a href="http://opengrok.github.com/OpenGrok/"
 title="Served by OpenGrok"><span id="fti"></span></a></p>
<p>Indexes created Mon Apr 10 12:05:06 EDT 2017</p>
</div></div>
</div>
<script type="text/javascript" src="/1830srosviews/js/jquery-3.2.0.min.js"></script>
<script type="text/javascript" src="/1830srosviews/js/jquery-ui-1.12.0-custom.min.js"></script>
<script type="text/javascript" src="/1830srosviews/js/jquery.tablesorter-2.26.6.min.js"></script>
<script type="text/javascript" src="/1830srosviews/js/tablesorter.parsers-0.0.1.js"></script>
<script type="text/javascript" src="/1830srosviews/js/searchable-option-list-2.0.3.min.js"></script>

<script type="text/javascript" src="/1830srosviews/js/utils-0.0.8.js"></script>
</body>
</html>

```

### PR DESCRIPTION
**Pull request by vladak**
fixes #1533

### CODE REVIEW

File: src/org/opensolaris/opengrok/index/Filter.java [hunk 1]
```diff
@@ -102,47 +102,58 @@ public void clear() {
...
+                if (absolute.endsWith(path)) {
+                    return true;
+                }
             }
         }
 
         return false;
     }
 
+    public boolean match(File file) {
```
Comment (tulinkry) [inline src/org/opensolaris/opengrok/index/Filter.java]: a javadoc would be nice

File: src/org/opensolaris/opengrok/index/Filter.java [hunk 2]
```diff
@@ -102,47 +102,58 @@ public void clear() {
     /**
```
Comment (tulinkry) [inline src/org/opensolaris/opengrok/index/Filter.java]: File /home/ktulinge/workspace/OpenGrok/src/org/opensolaris/opengrok/index/Filter.java has been changed but contains an old copyright. (expected:<2017> but was:<2016>)

Comment (vladak) [inline src/org/opensolaris/opengrok/index/Filter.java]: fixed

File: src/org/opensolaris/opengrok/index/Filter.java (hunk 1 above)
Comment (vladak) [inline src/org/opensolaris/opengrok/index/Filter.java]: fixed

**Pull request merged**